libs/contrib/Data/Nat/Order/Properties.idr

```diff
@@ -0,0 +1,100 @@
+||| Additional properties/lemmata of Nats involving order
+module Data.Nat.Order.Properties
+
+import Syntax.PreorderReasoning
+import Syntax.PreorderReasoning.Generic
+import Data.Nat
+import Data.Nat.Order
+import Data.Nat.Order.Strict
+import Decidable.Equality
+import Decidable.Order
+import Decidable.Order.Strict
+import Data.Bool.Decidable
+
+
+%default total
+export
+LTESuccInjectiveMonotone : (m, n : Nat) -> Reflects (m `LTE` n) b -> Reflects (S m `LTE` S n) b
+LTESuccInjectiveMonotone m n (RTrue      m_lte_n) = RTrue  $ LTESucc m_lte_n
+LTESuccInjectiveMonotone m n (RFalse not_m_lte_n) = RFalse $ \case 
+                                                               LTESucc m_lte_n => not_m_lte_n m_lte_n
+                                                               
+export
+lteReflection : (a, b : Nat) -> Reflects (a `LTE` b) (a `lte` b)
+lteReflection 0 b = RTrue LTEZero
+lteReflection (S k) 0 = RFalse \sk_lte_z => absurd sk_lte_z
+lteReflection (S a) (S b) = LTESuccInjectiveMonotone a b (lteReflection a b)
+
+-- For example:
+export
+lteIsLTE : (a, b : Nat) -> a `lte` b = True -> a `LTE` b
+lteIsLTE  a b prf = invert (replace {p = Reflects (a `LTE` b)} prf (lteReflection a b))
+
+export
+notlteIsLT : (a, b : Nat) -> a `lte` b = False -> b `LT` a
+notlteIsLT a b prf = notLTImpliesGTE
+                       \prf' =>
+                         (invert $ replace {p = Reflects (S a `LTE` S b)} prf 
+                                 $ lteReflection (S a) (S b)) prf'
+  
+
+
+
+-- The converse to lteIsLTE:
+export
+LteIslte : (a, b : Nat) -> a `LTE` b -> a `lte` b = True
```

Comment:
This is a direct consequence of the `Reflects` proof.
You should be able to prove a general theorem:

```idris
Reflects p b -> if c then p else Not p -> b = c
````

giving you both `LTE a b -> lte a b = True` and `Not (LTE a b) -> lte a b = False`

Reply:
Done, thanks!